performance/src/main/java/org/mule/test/SorterLargeAppBenchmark.java

```diff
@@ -0,0 +1,162 @@
+      ObjectStoreManager.class,
+      ExpressionLanguageExtension.class,
+      ExpressionLanguage.class,
+      QueueManager.class,
+      StreamingManager.class,
+      ConfigurationProvider.class,
+      Config.class,
+      SecurityManager.class,
+      FlowConstruct.class,
+      MuleConfiguration.class,
+      Initialisable.class
+  };
+
+  @Override
+  protected String getConfigFile() {
+    return "benchmark/large-app.xml";
+  }
+
+  @Override
+  protected boolean isStartContext() {
+    return false;
+  }
+
+  @Override
+  protected boolean doTestClassInjection() {
+    return true;
+  }
+
+  @Setup
+  public void setUp() throws Throwable {
+    setUpMuleContext();
+
+    // Use Java Reflection API in order to get the SpringRegistry
+    MuleRegistry muleRegistry = ((MuleContextWithRegistry) muleContext).getRegistry();
+    Field registryField = muleRegistry.getClass().getDeclaredField("registry");
+    registryField.setAccessible(true);
+    SpringRegistry springRegistry = (SpringRegistry) registryField.get(muleRegistry);
+
+    declaredDependencyResolver = new DeclaredDependencyResolver(springRegistry);
+    autoDiscoveredDependencyResolver = new AutoDiscoveredDependencyResolver(springRegistry);
+    configurationDependencyResolver = springRegistry.getConfigurationDependencyResolver();
+
+    resolver = new DependencyGraphBeanDependencyResolver(configurationDependencyResolver, declaredDependencyResolver,
+                                                         autoDiscoveredDependencyResolver, springRegistry);
+    graphSorter = new DependencyGraphLifecycleObjectSorter(resolver, allowedTypes);
+    springSorter = new SpringLifecycleObjectSorter(allowedTypes, springRegistry);
+
+    lookupObjects = springRegistry.lookupByType(Object.class);
+    lookupObjectNames = new ArrayList<>();
+
+  }
+
+  @Benchmark
+  @BenchmarkMode(AverageTime)
+  public List<Object> largeAppGraphSorter() {
+    // add objects with a dependency graph sorter
+    lookupObjects.forEach((key, value) -> {
+      graphSorter.addObject(key, value);
+      lookupObjectNames.add(key);
+    });
+    graphSorter.setLifeCycleObjectNameOrder(lookupObjectNames);
+
+    // sort vertices
+    return graphSorter.getSortedObjects();
+  }
+
+  @Benchmark
+  @BenchmarkMode(AverageTime)
+  public List<Object> largeAppSpringSorter() {
+    // add objects with a spring sorter (deprecated)
+    lookupObjects.forEach((key, value) -> {
+      springSorter.addObject(key, value);
+      lookupObjectNames.add(key);
+    });
+    springSorter.setLifeCycleObjectNameOrder(lookupObjectNames);
+
+    // sort vertices
+    return springSorter.getSortedObjects();
+  }
+
+  @Benchmark
+  @BenchmarkMode(AverageTime)
+  public List<Object> largeAppGraphSorterGetSortedObjectsOnly() {
+    return graphSorter.getSortedObjects();
+  }
+
+  @Benchmark
+  @BenchmarkMode(AverageTime)
+  public List<Object> largeAppSpringSorterGetSortedObjectsOnly() {
+    return springSorter.getSortedObjects();
+  }
```

Comment:
I think we would need the benchmark for the other part of the test (the adding of the objects) as well to get the proper times for the `getSortedObjects` alone

Reply:
@elrodro83 yes, we have three benchmarks per sorter (graphSorter and springSorter). 
1. benchmark for the adding objects(building graphs) part
2. benchmark for the get sorted objects (sorting objects from all the graphs we built) part
3. benchmark for both

Also, this version instantiates the sorter in the benchmark method instead of the set up. In this way, we don't reuse the same sorter multiple times that was created during the set up. The sorter that was instantiated in the set up will be only used when we measure this part.(2. get sorted objects (sorting objects from all the graphs we built) part)